commit 050a036dda1e91dc688616e20db9a196d2950b1c
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 16:10:06 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-tar-cJf.diff

diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index 17b1cd192..14e9b9f4a 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -2830,7 +2830,8 @@ proc extract_archive_metadata {archive_location archive_type metadata_types} {
             set raw_contents [exec -ignorestderr [findBinary tar ${portutil::autoconf::tar_path}] -xO${qflag}f $archive_location ./+CONTENTS]
         }
         txz {
-            set raw_contents [exec -ignorestderr [findBinary tar ${portutil::autoconf::tar_path}] -xO${qflag}f $archive_location --use-compress-program [findBinary xz ""] ./+CONTENTS]
+#            set raw_contents [exec -ignorestderr [findBinary tar ${portutil::autoconf::tar_path}] -xO${qflag}f $archive_location --use-compress-program [findBinary xz ""] ./+CONTENTS]
+            set raw_contents [exec -ignorestderr [findBinary tar ${portutil::autoconf::tar_path}] -xOJ${qflag}f $archive_location ./+CONTENTS]
         }
         tlz {
             set raw_contents [exec -ignorestderr [findBinary tar ${portutil::autoconf::tar_path}] -xO${qflag}f $archive_location --use-compress-program [findBinary lzma ""] ./+CONTENTS]
